server/client_test.go

```diff
@@ -485,6 +486,134 @@ func TestClientPubWithQueueSub(t *testing.T) {
 	}
 }
 
+func TestQueueSubscribePermissions(t *testing.T) {
+	cases := []struct {
+		name    string
+		perms   *SubjectPermission
+		subject string
+		queue   string
+		want    string
+	}{
+		{
+			name:    "plain subscription on foo",
+			perms:   &SubjectPermission{Allow: []string{"foo"}},
+			subject: "foo",
+			want:    "+OK\r\n",
+		},
+		{
+			name:    "queue subscribe with allowed group",
+			perms:   &SubjectPermission{Allow: []string{"foo bar"}},
+			subject: "foo",
+			queue:   "bar",
+			want:    "+OK\r\n",
+		},
+		{
+			name:    "queue subscribe with wildcard allowed group",
+			perms:   &SubjectPermission{Allow: []string{"foo bar.*"}},
+			subject: "foo",
+			queue:   "bar.fizz",
+			want:    "+OK\r\n",
+		},
+		{
+			name:    "queue subscribe with full wildcard subject and subgroup",
+			perms:   &SubjectPermission{Allow: []string{"> bar.>"}},
+			subject: "whizz",
+			queue:   "bar.bang",
+			want:    "+OK\r\n",
+		},
+		{
+			name:    "plain subscribe with full wildcard subject and subgroup",
+			perms:   &SubjectPermission{Allow: []string{"> bar.>"}},
+			subject: "whizz",
+			want:    "-ERR 'Permissions Violation for Subscription to \"whizz\"'\r\n",
+		},
+		{
+			name:    "deny plain subscription on foo",
+			perms:   &SubjectPermission{Allow: []string{">"}, Deny: []string{"foo"}},
+			subject: "foo",
```

Comment:
I think we should have added a `queue` in this test, currently only testing plain subscription permissions /cc @variadico